internal/ethapi/api.go

```diff
@@ -795,46 +825,6 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNrOrHash rpc.Blo
 			}
 		}
 	}
-	// Set default gas & gas price if none were set
-	gas = uint64(math.MaxUint64 / 2)
-	if args.Gas != nil {
-		gas = uint64(*args.Gas)
-	}
-	if globalGasCap != nil && globalGasCap.Uint64() < gas {
-		log.Warn("Caller gas above allowance, capping", "requested", gas, "cap", globalGasCap)
-		gas = globalGasCap.Uint64()
-	}
-
-	// Create new call message
-	gasPrice := new(big.Int)
-	if args.GasPrice != nil {
-		gasPrice = args.GasPrice.ToInt()
-	}
-
-	// Checking against 0 is a hack to allow users to bypass the default gas price being set by web3,
-	// which will always be in Gold. This allows the default price to be set for the proper currency.
-	// TODO(asa): Remove this once this is handled in the Provider.
```

Comment:
@asaj Has this been handled in the provider / does the new gas price logic make sense?

Reply:
To clarify, this doesn't affect gas estimation, because for gas estimation we were previously setting the gas price to zero in `core.ApplyEstimatorMessage`, and after this change we are setting it to zero below in `DoEstimateGas`.  

The behavior change is as follows (copied from the PR notes):
[copy/pasted]
To sum up, our old behavior was:
* for eth_estimateGas, set gas price and price gas minimum to zero, and set account balance to maxuint
* for eth_call, if gas price arg isn't passed in or is zero, set it to suggested gas price, and set account balance to maxuint

The new behavior is:
* for eth_estimateGas, set gas price and gas price minimum to zero
* for eth_call, set gas price minimum to zero

[end copy/pasted]

My understanding is that in the provider we set the gas price to zero, so that these changes won't cause any problems. It would also not cause problems if the gas price is null (which is interpreted as being zero).

The only cases where it would make a difference are:
1.  If you specify a non-zero gas price in CELO and your account doesn't have enough funds to pay for the gas.  In that case, the pre-PR behavior would have allowed for a successful call (because of the hack from upstream of setting the balance to maxuint).  Post-PR, the call would say the account doesn't have enough funds to pay the fee.
2. If the call itself somehow depends on the account's balance, in which case the pre-PR result would have been incorrect (due to the faking of the balance as maxuint) and the post-PR result would be correct.